h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -151,8 +168,51 @@ public void setGroupBy(ArrayList<Expression> group) {
         return group;
     }
 
-    public HashMap<Expression, Object> getCurrentGroup() {
-        return currentGroup;
+    /**
+     * Is there currently a group-by active
+     */
+    public boolean isCurrentGroup() {
+        return currentGroupByExprData != null;
+    }
+
+    /**
+     * Get the group-by data for the current group and the passed in expression.
+     */
+    public Object getCurrentGroupExprData(Expression expr) {
+        Integer index = exprToIndexInGroupByData.get(expr);
+        if (index == null) {
+            return null;
+        }
+        cachedLookup = expr;
+        cachedLookupIndex = index;
+        return currentGroupByExprData[index];
+    }
+
+    /**
+     * Set the group-by data for the current group and the passed in expression.
+     */
+    public void setCurrentGroupExprData(Expression expr, Object obj) {
+        if (cachedLookup == expr) {
+            assert currentGroupByExprData[cachedLookupIndex] != null;
```

Comment:
I got an assertion error here in the original test case.

Reply:
The logic is simple enough. Aggregate (or something) called `getCurrentGroupExprData()`, got a `null` value and sets a newly allocated `AggregateData` with `setCurrentGroupExprData()`. So `currentGroupByExprData[cachedLookupIndex]` can legally be `null` here.

Reply:
Even more, it should be `null` here. Assertion should be reversed.